src/compressed_chunk.c

```diff
@@ -466,27 +535,33 @@ static void Compressed_Serialize(Chunk_t *chunk,
                                  SaveStringBufferFunc saveStringBuffer) {
     CompressedChunk *compchunk = chunk;
 
-    saveUnsigned(ctx, compchunk->size);
+    saveUnsigned(ctx, compchunk->size_ts);
+    saveUnsigned(ctx, compchunk->size_values);
     saveUnsigned(ctx, compchunk->count);
-    saveUnsigned(ctx, compchunk->idx);
+    saveUnsigned(ctx, compchunk->idx_ts);
+    saveUnsigned(ctx, compchunk->idx_values);
     saveUnsigned(ctx, compchunk->baseValue.u);
     saveUnsigned(ctx, compchunk->baseTimestamp);
     saveUnsigned(ctx, compchunk->prevTimestamp);
     saveUnsigned(ctx, compchunk->prevTimestampDelta);
     saveUnsigned(ctx, compchunk->prevValue.u);
     saveUnsigned(ctx, compchunk->prevLeading);
     saveUnsigned(ctx, compchunk->prevTrailing);
-    saveStringBuffer(ctx, (char *)compchunk->data, compchunk->size);
+    saveStringBuffer(ctx, (char *)compchunk->data_ts, compchunk->size_ts);
+    saveStringBuffer(ctx, (char *)compchunk->data_values, compchunk->size_values);
 }
 
 #define COMPRESSED_DESERIALIZE(chunk, ctx, readUnsigned, readStringBuffer, ...)                    \
     do {                                                                                           \
         CompressedChunk *compchunk = (CompressedChunk *)malloc(sizeof(*compchunk));                \
                                                                                                    \
-        compchunk->data = NULL;                                                                    \
-        compchunk->size = readUnsigned(ctx, ##__VA_ARGS__);                                        \
+        compchunk->data_ts = NULL;                                                                 \
+        compchunk->data_values = NULL;                                                             \
+        compchunk->size_ts = readUnsigned(ctx, ##__VA_ARGS__);                                        \
+        compchunk->size_values = readUnsigned(ctx, ##__VA_ARGS__);                                        \
         compchunk->count = readUnsigned(ctx, ##__VA_ARGS__);                                       \
-        compchunk->idx = readUnsigned(ctx, ##__VA_ARGS__);                                         \
+        compchunk->idx_ts = readUnsigned(ctx, ##__VA_ARGS__);                                      \
+        compchunk->idx_values = readUnsigned(ctx, ##__VA_ARGS__);                                  \
```

Comment:
in rdb.h need to increment the rdb version:
https://github.com/RedisTimeSeries/RedisTimeSeries/blob/fd31e3e65ad77c991ae2462317fccf5b555cd469/src/rdb.h#L21
And deserialize the values by the enc_ver, need to pass it here:
https://github.com/RedisTimeSeries/RedisTimeSeries/blob/bd23b1e07f766819b55d4512c3e8e63e1f556121/src/rdb.c#L122

Reply:
I have been thinking that it seems to me, that we cannot be backwards compatible with older versions of the RDB. 
This is because, we have fundamentally changed the chunk data structure. Thus, even if we I check for the RDB version, I cannot deserialize the data into 2 buffers, if it was serialized into 1 buffer. 
All the functions operating on the data now expect 2 buffers.

Reply:
I think the first option, to handle old rdbs and normalize their data to the new architecture of 2 separate.
In the future when the old versions will be out of life we can remove this logic. 

@danni-m, @filipecosta90  Would like to hear your input here.